### PR TITLE
Json

### PR DESCRIPTION
**Pull request by aaronjridley**
# Description

Addresses # 39

This makes the aether.in file into a json format file.  Instead of having a #feature it uses the json format for inputs.  This also allows used to add new parameters into the code without ever actually having to add any code for the file input, which is a nice side benefit.  It adds the dependence on the nlohmann json headers, but this dependence was already included.

Also, I cleaned up a bunch of files using the AStyle code.  This should have done for the Electrodynamics pull request, but wasn't fully completed in that.  There are a few files where just spaces and such are changed.

## Type of change

- Breaking change (fix or feature that would cause existing functionality
  to not work as expected)

# How Has This Been Tested?

clones the json branch from github from scratch, checkouted out the code, and ran it.  It works as expected!

## Test configuration

* Operating system: mac OSX
* Compiler, version number: gcc version 10

# Checklist:

- [N/A] Make sure you are merging into the ``develop`` (not ``master``) branch
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [N/A] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: share/run/aether.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+{
+
+    "Debug" : {
+	"iVerbose" : 0,
+	"dt" : 10.0},
+
+    "Mysetting" : "blerg",
```
Comment (aburrell) [inline share/run/aether.json:8]: What is "blerg"?

File: src/read_input_file.cpp [hunk 2]
```diff
@@ -1,222 +1,62 @@
...
-#include <vector>
-#include <string>
-#include <iostream>
-#include <fstream>
-
 #include "aether.h"
 
 // -----------------------------------------------------------------------
-// Read input file!
+// Read json file
```
Comment (aburrell) [inline src/read_input_file.cpp:7]:
```suggestion
// Read the json input file
```

File: share/run/aether.json (hunk 1 above)
Comment (aaronjridley) [reply share/run/aether.json]: I change the name of the variable to be a bit more clear - the user can add whatever variables they want and actually use them in the code.  So, this is an example.  I added a better example.

File: src/read_input_file.cpp (hunk 2 above)
Comment (aaronjridley) [reply src/read_input_file.cpp]: Made a comment change and committed.

**Comment by aaronjridley** [comment]
Are you good with this @aburrell?

**Comment by aburrell** [comment]
I haven't had a chance to look at it yet.  It's on my list.

**Review by Ionospheres — COMMENTED** [review]
From a functionality standpoint this looks great. I got it to run right out of the box and the output parameters are reasonable. 
1) Is there documentation yet/somewhere for what variables can be sensibly used in the aether.json run file? 
2) Do we want a message for when a nonsense variable is used in the aether.json file? To alert for potential typos or sillyness? I gave it  "Potato" : [35748], and it was just ignored without warnings.

**Comment by aaronjridley** [comment]
1. I need to make some documentation for this.  I have not.  This should go into a manual, which should be on overleaf or something.  We have a document started there, but I have not done anything with it in a bit. The TL;DR for this is that all variables should be defined in the UA/inputs/defaults.json file. They should all be there.  Then, the aether.json file is a perturbation on this - only the things that you want to change need to be added here (or copied from the defaults.json file and edited.) (I should also say that there was no documentation on the OLD input files, so this merge request was not meant to get us to a BETTER place, it was meant to get us to a similar place....)

2a. I have debated this.  The super cool thing about the json system is that you can add your own variables (like what you have there), and you can USE THEM IN THE CODE! This is because the code just reads in the json file and stores it.  You then can access the data with some methods that I implemented, and a developer can just start using their own inputs without having to make code modification in the input.h file.  This would be for testing new things, like you want a new coefficient or something for your own section of code, you can just start using that variable.  Once you want to merge it into the develop branch, then it should be incorporated into inputs.h and inputs.cpp. But, this gives users / developers a lot of flexibility.

2b.  In theory, we should cycle through all of the required variables and throw a message if someone is missing one or if someone has something that is not correct. Again, this is added functionality and good for a beta-release, but I don't think that we are to alpha yet, so I would definitely put this as an Issue for future development.

**Comment by Ionospheres** [comment]
1) Agreed. In the meantime, this sounds like a good and quick thing to add to the ReadMe so new users of the json way will know where to look for the list of default variable names they can use to customize their run.

2) I really like the flexibility of the json system. If not now, then definitely as a future feature we should add some warning statement that notifies the user that a variable is not in the default list. At a minimum it would help users track down simple typing mistakes from fumble fingers.

**Comment by aaronjridley** [comment]
I have modified the README.md file to describe the two json files.

**Comment by aaronjridley** [comment]
Added issue on input checker.

**Review by aburrell — APPROVED** [review]

**Pull request merged**